x-pack/elastic-agent/pkg/agent/application/reexec/manager.go

```diff
@@ -30,14 +30,19 @@ type ExecManager interface {
 	ShutdownComplete()
 }
 
-// Manager returns the global reexec manager.
-func Manager(log *logger.Logger, exec string) ExecManager {
+// NewManager returns the global reexec manager.
+func NewManager(log *logger.Logger, exec string) ExecManager {
 	execSingletonOnce.Do(func() {
 		execSingleton = newManager(log, exec)
 	})
 	return execSingleton
 }
 
+// Manager returns the global reexec manager.
+func Manager() ExecManager {
+	return execSingleton
```

Comment:
This does not look like a good idea why not having just one method with Once-initialization

Reply:
I would like to, but that deep into the code, I didn't want to have to pass in the logger and the path to the executable. I wanted that to be set at startup and later usages of the manager, can get the singleton.